### PR TITLE
Add logit_bias to Flash Attention Kernels

### PR DESCRIPTION
**Pull request by HahTK**
This requires compiler with FA Kernel that supports logit_bias.
It has been implemented ton the ENABLED_NEW_UNSHARDED_ATTN_KERNEL only.
If we need this for a legacy kernel, then a request should be filed to the compiler team.

Upstream code has changed w.r.t to logit_bias. Please see comments in this code related to upstrearming.

### CODE REVIEW

File: axlearn/common/flash_attention/neuron_attention.py [hunk 1]
```diff
@@ -57,8 +60,10 @@ def _mha_forward(query, key, value, causal, softmax_scale):
       import neuronxcc.nki.language as nl
 
       assert (num_heads % 2) == 0 and (num_heads // 2 > 0), f'unexpect num_heads: {num_heads}'
-      attn_output, lse = flash_fwd[batch_size, nl.nc(2) * (num_heads//2)](q, k, v, seed, use_causal_mask=causal, softmax_scale=softmax_scale, mixed_precision=True, dropout_p=0.0)
+      attn_output, lse = flash_fwd[batch_size, vnc(2), num_heads//2](q, k, v, bias, seed, use_causal_mask=causal, softmax_scale=softmax_scale, mixed_precision=True, dropout_p=0.0)
```
Comment (indhub) [inline axlearn/common/flash_attention/neuron_attention.py]: why nl.nc(2) => vnc(2)?

File: axlearn/common/flash_attention/neuron_attention.py [hunk 2]
```diff
@@ -95,8 +100,10 @@ def _mha_backward(causal, softmax_scale, res, d_attn_output):
       from neuronxcc.nki.kernels.attention import flash_attn_bwd
       import neuronxcc.nki.language as nl
       assert (num_heads % 2) == 0 and (num_heads // 2 > 0), f'unexpected num_heads: {num_heads}'
-      d_query, d_key, d_value = flash_attn_bwd[batch_size, nl.nc(2) * (num_heads//2)](q, k, v, o, dy, lse, seed, use_causal_mask=causal, mixed_precision=True, dropout_p=0.0, softmax_scale=softmax_scale)
+      d_query, d_key, d_value = flash_attn_bwd[batch_size, vnc(2), num_heads // 2](q, k, v, o, dy, lse, seed, bias, use_causal_mask=causal, mixed_precision=True, dropout_p=0.0, softmax_scale=softmax_scale)
```
Comment (indhub) [inline axlearn/common/flash_attention/neuron_attention.py]: why nl.nc(2) => vnc(2)?

File: axlearn/common/flash_attention/neuron_attention.py (hunk 1 above)
Comment (HahTK) [reply axlearn/common/flash_attention/neuron_attention.py]: seems like ships passing in the night. Both old and new code was vnc(2) when i pulled and tested. It got changed and was missed in merge conflict resolution. PR to fix in a sec

File: axlearn/common/flash_attention/neuron_attention.py (hunk 2 above)
Comment (HahTK) [reply axlearn/common/flash_attention/neuron_attention.py]: seems like ships passing in the night. Both old and new code was vnc(2) when i pulled and tested. It got changed and was missed in merge conflict resolution. PR to fix in a sec

**Pull request closed without merging**